### ISSUE TITLE
[Android] Intermittently Invalid Brave Rewards Wallet ("Rewards profile not yet created" even though has already been created)

### ISSUE THREAD

**Issue by jonathansampson** [brave/brave-browser#21057]
## Description

The Rewards Wallet on Android appears to intermittently enter into an invalid state for some users. Interruptions with Brave Ads (confirmations failing) are observed in addition to this, and are likely related.

## Steps to reproduce

TBD

## Actual result

The wallet fails to load, and/or is believed to be invalid.

When this happens brave://rewards-internals/ will state that the Wallet has not yet been created. Restarting the application may resolve this for some users (possible race condition?). Verbose Logs will also record a failure to redeem unblinded tokens due to _an invalid wallet_.

## Issue reproduces how often

Intermittently. Frequency TBD.

## Version/Channel Information:

Observed on multiple releases, but most recently on 1.35.100.

## Device details
   - Device type (Phone, Tablet, Phablet): Pixel 5
   - Android version: 12
   
## Brave version

1.35.100

### Additional information

Potentially related to other issues, including #18717 and #13026.

**Comment by newuser-web** [comment]
Similar issue was reported related to Brave News. #19800

**Comment by newuser-web** [comment]
Similar issue were reported related to Brave News and NTP sponsored image #17352 #19800

**Comment by srirambv** [comment]
This seems to be same as #13489. Ads show up but may not be redeemed due to wallet showing as invalid. Disabling ads and reenabling fixes it but may remove the custodian linkage. 

cc: @Miyayes please look at the linked issues and close them as needed

**Comment by jonathansampson** [comment]
@srirambv In my case, disabling ads resulted in the inability to turn them back on. I had to restart the mobile browser to resolve the issue.

**Comment by srirambv** [comment]
> disabling ads resulted in the inability to turn them back on.

I think you can just close the rewards page and reopen another one and enable it. I don't think a browser restart is required. But yes I have seen this happen as well. I have been running into this issue for a very long time and still not able to figure out whats causing the issue.

**Comment by Miyayes** [comment]
@srirambv When the Rewards wallet info shows up again, do you know if it is the _same_ Wallet Payment ID as before, or was a _new_ one written? Just want to doublecheck.

**Comment by Miyayes** [comment]
@Brave-Matt just reproduced this on his Android device.

**Comment by Miyayes** [comment]
Seems like frequency of this is increasing. Will increase priority.

**Comment by Miyayes** [comment]
Seems like this hasn't been happening in more recent versions of Brave. May have resolved itself, but will keep issue open for awhile longer and keep monitoring for reports.

**Comment by 424344** [comment]
unfortunately it's not resolved. happened tom me yesterday.
I agree that currently it seems less frequent than before, but it's still not fixed.

**Comment by Miyayes** [comment]
> unfortunately it's not resolved. happened tom me yesterday. I agree that currently it seems less frequent than before, but it's still not fixed.

@424344 Can you confirm which version of Brave app you are on when you see this happen again?

**Comment by 424344** [comment]
I'm not sure about the version, but I'm always up to date with the release in the play store (right now it's v1.41.96)
Probably last time it happened I was running v1.40.x

**Comment by Uni-verse** [comment]
Reproduced using version `1.43.86`

```
09-03 01:28:34.002 26586 26586 I chromium: [INFO:redeem_unblinded_payment_tokens.cc(53)] Failed to redeem unblinded payment tokens due to invalid wallet
09-03 01:28:34.002 26586 26586 V chromium: [VERBOSE1:redeem_unblinded_payment_tokens.cc(148)] Failed to redeem unblinded payment tokens
```

**Comment by Miyayes** [comment]
> Reproduced using version `1.43.86`
> 
> ```
> 09-03 01:28:34.002 26586 26586 I chromium: [INFO:redeem_unblinded_payment_tokens.cc(53)] Failed to redeem unblinded payment tokens due to invalid wallet
> 09-03 01:28:34.002 26586 26586 V chromium: [VERBOSE1:redeem_unblinded_payment_tokens.cc(148)] Failed to redeem unblinded payment tokens
> ```

Do you have some information about the Android device and its particular configuration/state? Thanks.

**Comment by TMTRTLP** [comment]
Yeah, I believe I'm running into the same issue right now.

Pixel 6a - November Security Patch

![Screenshot_20221206-201237~2](https://user-images.githubusercontent.com/16298909/206071209-dc3e29f4-0b81-4446-a3b8-bcfd50977213.png)
![Screenshot_20221206-200817](https://user-images.githubusercontent.com/16298909/206071217-134b45b7-7ecf-4425-90c3-ee8851528d68.png)

**Comment by Miyayes** [comment]
Report from user who runs into this regularly:

>I’ve previously reported this happens (months ago when the rewards were all paying out wrong) yes, and yes this also fixes itself when the browser relaunch works. The only difference with mine is I cannot see the balance just a whirring wheel, whereas I noted in the screenshot above they can at least see their balance so perhaps not the same issue after all.
Nothing unique, standard set up no modifications or VPN - Samsung A50
Android phone running Brave version 1.46.146, Chromium 108.0.53591.128.
Yes I can but timing is important, I cannot do this during work hours

https://community.brave.com/t/why-is-it-showing-unverified/461044/6?u=chriscat

**Comment by emerick** [comment]
Sample failure log (from @deeppandya):

[brave_rewards_log.txt](https://github.com/brave/brave-browser/files/10902038/brave_rewards_log.txt)

**Comment by Miyayes** [comment]
This issue may have been resolved as part of 1.48.x., so will keep an eye out for more support reports, and if there are any more reports here.

**Comment by Miyayes** [comment]
Still seeing a lot of reports of this. On Android, users need to force kill the app or restart their phone.

Some symptoms are custodian selection buttons not working (unresponsive).
